File: Leetcode/509. Fibonacci Number/fibonacci_number.py
class Solution:
    def fib(self, n: int) -> int:
        if n < 2:
            return n
        return self.fib(n-1)+self.fib(n-2)

    def fib1(self, n: int) -> int:
        if n < 2:
            return n
        arr = [0, 1]
        for i in range(2, n+1):
            arr.append(arr[i-1]+arr[i-2])
        return arr[n]


item = Solution()
print(item.fib(6))
print(item.fib1(6))
# 6

File: Leetcode/509. Fibonacci Number/fibonacci_number.rs
struct Solution {}

impl Solution {
    pub fn fib(n: i32) -> i32 {
        if n<2{
            return n
        }
        return Self::fib(n-1)+Self::fib(n-2)
    }
    pub fn fib1(n: i32) -> i32 {
        if n<2{
            return n;
        }
        let mut values = vec![0,1];
        for i in 2..n+1 {
            values.push(values[i as usize-1]+values[i as usize -2])
        }
        return values[n as usize]
    }
}
fn main() {
    let res = Solution::fib(5);
    println!("{:?}", res);
    let res1 = Solution::fib1(5);
    println!("{:?}", res1);
    // res -> 5
}
